### ISSUE TITLE
Converting Datetime to Natural Language produces TextFeaturizer error during pipeline fit

### ISSUE THREAD

**Issue by dsherry** [alteryx/evalml#2296]
**Reproducer**
```python
import evalml
import pandas as pd
import woodwork as ww
import pytest

df = pd.read_csv('/Users/dylan.sherry/Downloads/fraud_detection_cleaned_v2_2500rows.csv')
dt = ww.DataTable(df)
y = dt.pop('fraud')
assert str(dt.logical_types['datetime']) == 'Datetime'

# manually change datetime to natural language. strange thing to do but it is possible.
dt = dt.set_types(logical_types={'datetime': 'NaturalLanguage'})
assert str(dt.logical_types['datetime']) == 'NaturalLanguage'

# define a pipeline _without_ a datetime featurizer, since there's no datetime-typed features in the data
pipeline = evalml.pipelines.BinaryClassificationPipeline([
    'Imputer', 'One Hot Encoder', 'Text Featurization Component', 'Elastic Net Classifier'])

# get error during pipeline fit
with pytest.raises(TypeError, match='invalid type promotion'):
    pipeline.fit(dt, y)
```

**Root Cause**
In our `TextFeaturizer`, we normalize to woodwork DataTable, get a pandas dataframe, run featuretools to compute some features, then run our LSA component to produce more features.

When we run the LSA component, we should be passing along the woodwork DataTable, because that contains the information that the "datetime" column happens to be NaturalLanguage-typed. However, instead we currently pass the pandas dataframe. This means the LSA component will run woodwork type inference, determine that "datetime" is a Datetime-typed feature, try to use that feature to generate TF-IDF features and error out.

[Here's the line](https://github.com/alteryx/evalml/blob/main/evalml/pipelines/components/transformers/preprocessing/text_featurizer.py#L122) which causes the problem.

**Fix**
Short-term: update that line to provide the woodwork DataTable instead of the pandas dataframe, to preserve the user-specified feature types.
Long-term: hey @freddyaboulton does your woodwork accessor feature branch fix this problem? If not, we should definitely make a fix on there as well.

**Comment by freddyaboulton** [comment]
@dsherry I think there are two issues here! The issue you pointed out with the TextFeaturizer implementation is valid but I don't think it's what's causing the error in your repro. If I instantiate a TextFeaturizer and run it with your data, I get a different stacktrace (using fraud cause I don't have your dataset):

```python
import evalml
import pandas as pd
from evalml.pipelines.components import TextFeaturizer
import pytest

X, y = evalml.demos.load_fraud(1000)

assert str(X.logical_types['datetime']) == 'Datetime'

X = X.set_types(logical_types={'datetime': 'NaturalLanguage'})

with pytest.raises(AttributeError, match="'Timestamp' object has no attribute 'lower'"):
    TextFeaturizer().fit_transform(X, y)
```

I think the other sneaky thing is that the `ComponentGraph` calls `pd.concat` and then `infer_feature_types` so the woodwork info gets nuked! [link](https://github.com/alteryx/evalml/blob/main/evalml/pipelines/component_graph.py#L324)

I think that's what actually happening in your repro because the `TextFeaturizer` doesn't even know there are text features in the data:

```python
import pytest

X, y = evalml.demos.load_fraud(1000)

X = X.set_types(logical_types={'datetime': 'NaturalLanguage', 'card_id': "Double", "store_id": "Double",
                               "amount": "Double"})
assert str(X.logical_types['datetime']) == 'NaturalLanguage'

# define a pipeline _without_ a datetime featurizer, since there's no datetime-typed features in the data
pipeline = evalml.pipelines.BinaryClassificationPipeline([
    'Imputer', 'One Hot Encoder', 'Text Featurization Component', 'Elastic Net Classifier'])
with pytest.raises(TypeError):
    pipeline.fit(X, y)

assert pipeline._component_graph[2]._text_columns == []
```

The first issue you bring up will be fixed by #2181 ! But the second issue is trickier. I'm hoping https://github.com/alteryx/woodwork/issues/884 will help. The other thing we can do is not use `concat` if there is only one `.x` input in the `ComponentGraph` but using ww.concat seems like the best thing to do.

**Comment by dsherry** [comment]
@freddyaboulton is gonna pick this up. Thank you @freddyaboulton !

Acceptance criterion: make sure each component in the graph is given the "right" feature types. In particular, if custom types were set which differ from woodwork's default inferences, those custom types should be what each component is given for that feature, not woodwork's default inference. Add unit test coverage which proves this.

We were just chatting about the code changes in #2297 . I think the woodwork accessor work fixed some of the issues there, which is good news. In particular, the text featurizer code looks much better now.
